cartography/intel/crxcavator/__init__.py

```diff
@@ -0,0 +1,35 @@
+import logging
+import os
+
+from cartography.intel.crxcavator.crxcavator import sync_extensions
+from cartography.util import run_cleanup_job
+
+logger = logging.getLogger(__name__)
+
+# API key for CRXcavator generaated from crxcavator.io portal
+CRXCAVATOR_API_KEY = os.environ.get('CREDENTIALS_CRXCAVATOR_API_KEY')
+
+# API for the CRXcavator API - https://api.crxcavator.io/v1 as of 07/09/19
+CRXCAVATOR_API_BASE_URL = os.environ.get('CRXCAVATOR_URL')
+
+
+def start_extension_ingestion(session, config):
+    """
+    If this module is configured, perform ingestion of CRXcavator data. Otherwise warn and exit
+    :param session: Neo4J session for database interface
+    :param config: Neo4J server URI and Update tag for data freshness
+    :return: None
+    """
+    if not CRXCAVATOR_API_BASE_URL or not CRXCAVATOR_API_KEY:
+        logger.debug('CRXcavator import is not configured - skipping this module. See docs to configure.')
+        return
+
+    common_job_parameters = {
+        "UPDATE_TAG": config.update_tag,
+    }
+    sync_extensions(session, common_job_parameters, CRXCAVATOR_API_KEY, CRXCAVATOR_API_BASE_URL)
```

Comment:
should we test the creds first and indicate that they are wrong to the user before moving to ingestion?

Reply:
@ecdavis is this something we want to do or let it move ingest and error?

Reply:
I don't think there's a need to check for cred validity twice.

This module is different from AWS and GCP in that it uses the `requests` library to make raw HTTP calls rather than using a client library.  `crxcavator.call_crxcavator_api()` does this and `data.raise_for_status()` [will raise an exception](https://2.python-requests.org//en/master/_modules/requests/models/#Response.raise_for_status) if the creds don't work.